src/Storefront/Resources/app/storefront/test/utility/loading-indicator/button-loading-indicator.util.test.js

```diff
@@ -52,12 +55,72 @@ describe('ButtonLoadingIndicatorUtil tests', () => {
         // Verify element removes loader and removes disabled class
         expect(buttonEl.disabled).toBeUndefined();
         expect(buttonEl.classList.contains('disabled')).toBe(false);
+        expect(buttonEl.classList.contains('is-loading-indicator-before')).toBe(false);
         expect(buttonEl.querySelector('.loader')).toBeNull();
     });
 
-    test('throws an error if parent element is not of type button or anchor', () => {
-        expect(() => {
-            new ButtonLoadingIndicatorUtil(document.querySelector('.illegal-button'));
-        }).toThrow('Parent element is not of type <button> or <a>');
+    test('replaces the button text with a loading indicator when the "inner" parameter is passed', () => {
```

Comment:
Should we also test the before and after option?

Reply:
The before option is already covered because it is the default. I can add a test for the "after" option.

Reply:
Done.